streams/src/main/java/org/apache/kafka/streams/processor/internals/GlobalStreamThread.java

```diff
@@ -259,19 +251,14 @@ void initialize() {
             for (final Map.Entry<TopicPartition, Long> entry : partitionOffsets.entrySet()) {
                 globalConsumer.seek(entry.getKey(), entry.getValue());
             }
-            lastFlush = time.milliseconds();
         }
 
         void pollAndUpdate() {
             final ConsumerRecords<byte[], byte[]> received = globalConsumer.poll(pollTime);
             for (final ConsumerRecord<byte[], byte[]> record : received) {
                 stateMaintainer.update(record);
             }
-            final long now = time.milliseconds();
-            if (now - flushInterval >= lastFlush) {
-                stateMaintainer.flushState();
-                lastFlush = now;
-            }
+            stateMaintainer.maybeCheckpoint();
```

Comment:
should we still keep an interval-based flush or is the delta check sufficient?

Reply:
For me, the idea is to only flush&checkpoint based on the delta of records, as we do for task state stores

Reply:
Thanks @cadonna for your comment. Yes, I agree with your point.
However, wouldn't that create a difference in behavior between global and local state stores ? I saw that flushing local state stores is based only on the delta. Please correct me if I am wrong

Reply:
No, checkpointing the local store is not only based on the delta. Method `maybeCheckpoint()` is called after a commit and a commit is done in regular time-intervals.
https://github.com/apache/kafka/blob/d144b7ee387308a59e52cbdabc7b66dd3b2926cc/streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java#L513